datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemAttachmentViewHolderFactory.kt

```diff
@@ -0,0 +1,387 @@
+        takePhoto.setOnClickListener { onTakePhotoClicked(questionnaireItem) }
+        selectFile.setOnClickListener { onSelectFileClicked(questionnaireItem) }
+        delete.setOnClickListener {
+          questionnaireItemViewItem.clearAnswer()
+          clearDeleteButton()
+          clearPhotoPreview()
+          clearFilePreview()
+        }
+      }
+
+      override fun displayValidationResult(validationResult: ValidationResult) {
+        when (validationResult) {
+          is NotValidated,
+          Valid -> error.visibility = View.GONE
+          is Invalid -> {
+            error.text = validationResult.getSingleStringValidationMessage()
+            error.visibility = View.VISIBLE
+          }
+        }
+      }
+
+      override fun setReadOnly(isReadOnly: Boolean) {
+        if (isReadOnly) {
+          takePhoto.isClickable = false
+          takePhoto.alpha = 0.6F
+          selectFile.isClickable = false
+          selectFile.alpha = 0.6F
+          delete.isClickable = false
+          delete.alpha = 0.6F
+        } else {
+          takePhoto.isClickable = true
+          takePhoto.alpha = 1F
+          selectFile.isClickable = true
+          selectFile.alpha = 1F
+          delete.isClickable = true
+          delete.alpha = 1F
+        }
+      }
+
+      private fun displayInitialAttachmentPreview() {
+        questionnaireItemViewItem.answers.firstOrNull()?.valueAttachment?.let { attachment ->
+          when (attachment.contentType.type) {
+            MimeType.IMAGE.value -> {
+              loadPhotoPreviewInBytes(attachment.data)
+              clearFilePreview()
+              loadDeleteButton(R.string.delete_image)
+            }
+            MimeType.DOCUMENT.value -> {
+              loadFilePreview(R.drawable.file, attachment.title)
+              clearPhotoPreview()
+              loadDeleteButton(R.string.delete_file)
+            }
+            MimeType.VIDEO.value -> {
+              loadFilePreview(R.drawable.video_file, attachment.title)
+              clearPhotoPreview()
+              loadDeleteButton(R.string.delete_file)
+            }
+            MimeType.AUDIO.value -> {
+              loadFilePreview(R.drawable.audio_file, attachment.title)
+              clearPhotoPreview()
+              loadDeleteButton(R.string.delete_file)
+            }
+          }
+        }
+      }
+
+      private fun displayActionButtons(questionnaireItem: QuestionnaireItemComponent) {
+        if (questionnaireItem.hasMimeTypeOnly(MimeType.IMAGE.value)) {
+          takePhoto.visibility = View.VISIBLE
+          loadUploadButton(R.drawable.image_file, R.string.upload_photo)
+        } else if (questionnaireItem.hasMimeType(MimeType.IMAGE.value)) {
+          takePhoto.visibility = View.VISIBLE
+          loadUploadButton(R.drawable.file, R.string.select_file)
+        } else {
+          takePhoto.visibility = View.GONE
+          loadUploadButton(R.drawable.file, R.string.select_file)
+        }
```

Comment:
i don't feel it's important to change the upload button icon depending on if image is the only mime type allowed. (@shelaghm fyi)

so we can actually just simplify this logic to set the visibility of take photo button depending on if image is allowed.

the upload button will always be the same. i think you can just get ride of the `loadUploadButton` function.

Reply:
okay

Reply:
@jingtang10 
I have an end result of this, I hope it's simple enough

```suggestion
      private fun displayTakePhotoButton(questionnaireItem: QuestionnaireItemComponent) {
        if (questionnaireItem.hasMimeType(MimeType.IMAGE.value)) {
          takePhotoButton.visibility = View.VISIBLE
        }
      }

      private fun displayUploadButton(questionnaireItem: QuestionnaireItemComponent) {
        when {
          questionnaireItem.hasMimeTypeOnly(MimeType.IMAGE.value) -> {
            uploadPhotoButton.visibility = View.VISIBLE
          }
          questionnaireItem.hasMimeTypeOnly(MimeType.AUDIO.value) -> {
            uploadAudioButton.visibility = View.VISIBLE
          }
          questionnaireItem.hasMimeTypeOnly(MimeType.VIDEO.value) -> {
            uploadVideoButton.visibility = View.VISIBLE
          }
          questionnaireItem.hasMimeTypeOnly(MimeType.DOCUMENT.value) -> {
            uploadDocumentButton.visibility = View.VISIBLE
          }
          else -> {
            uploadFileButton.visibility = View.VISIBLE
          }
        }
      }
```